src/patterns/InPlaceFilters/types.ts

```diff
@@ -0,0 +1,142 @@
+export type AuctionFilterButtonTypes = 'Filter' | 'Sort' | 'Sale' | 'Departments' | 'Month' | 'Location';
+
+/**
+ * Represents a single filter option (e.g., for a dropdown or button group)
+ */
+export type AuctionFilter = {
+  /** Unique identifier for the filter option */
+  id: string;
+  /** Display label for the filter option */
+  label: string;
+};
+
+/**
+ * Represents a filter button with a count and optional labels
+ */
+export type AuctionFilterButton = {
+  /** Number of items matching this filter */
+  auctionCount: number;
+  /** Optional list of filter labels for this button */
+  filterLabels?: AuctionFilter[];
+};
+
+/**
+ * Represents a single dimension/option within a filter (e.g., a checkbox or radio option)
+ */
+export type FilterDimension = {
+  /** Display label for the filter dimension */
+  label: string;
+  /** Whether this dimension is disabled */
+  disabled?: boolean;
+  /** Whether this dimension is currently active/selected */
+  active?: boolean;
+};
+
+/**
+ * Represents a filter group (e.g., "Sale", "Department") with its options
+ */
+export type FilterType = {
+  /** Display label for the filter group */
+  label: string;
+  /** Unique identifier for the filter group */
+  id: string;
+  /** Type of filter (e.g., 'checkbox', 'radio') */
+  type: string;
+  /** Set of filter dimensions/options for this filter group */
+  filterDimensions: Set<FilterDimension>;
+};
+
+/** Handler for when a filter input changes (checkbox, radio, or select) */
+type FilterChangeHandler = (e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>, filterId: string) => void;
+
+/** Handler for when filters are updated (e.g., after selection) */
+type FilterUpdateHandler = (returnCountOnly?: boolean) => void;
+
+/** Handler for clearing a filter group by its id */
+type ClearFilterHandler = (filterId: string) => void;
+
+/**
+ * Common props shared by all filter-related components
+ */
+interface BaseFilterProps {
+  /** Optional CSS class for the component */
+  className?: string;
+  /** Array of filter groups to display */
+  filters?: FilterType[];
+  /** Handler for filter selection changes */
+  handleFilterSelection?: FilterChangeHandler;
+  /** Handler for updating filters/results */
+  handleFilterUpdate?: FilterUpdateHandler;
+  /** Handler for clearing a filter group */
+  clearFilterUpdate?: ClearFilterHandler;
+  /** Number of results to display */
+  resultsCount?: number;
+}
+
+/**
+ * Optional aria-labels for accessibility on various filter button elements.
+ */
+export interface FilterButtonAriaLabels {
+  /** Aria-label for the filter button itself */
+  button?: string;
+  /** Aria-label for the drawer (mobile or filter drawer) */
+  drawer?: string;
```

Comment:
do we need different labels for the different breakpoints?  Whether they're a drawer or a popover they're really just one interactive element